src/libecalc/core/models/pump/pump.py

```diff
@@ -305,6 +309,16 @@ def evaluate_rate_ps_pd_density(
 
         # Allowed calculation points is where required head is less than or equal to actual pump head
         # and rates are less than or equal to maximum pump rate
+        pump_flags = np.where(
+            rates_m3_per_hour > self.pump_chart.maximum_rate, PumpFlag.INVALID_ABOVE_MAXIMUM_PUMP_RATE.value, pump_flags
+        )
+
+        pump_flags = np.where(
+            operational_heads > actual_heads + self._head_margin,
+            PumpFlag.INVALID_REQUIRED_HEAD_ABOVE_ACTUAL_HEAD.value,
+            pump_flags,
+        )
+
```

Comment:
PumpModelResults inherits is_valid from EnergyFunctionResult and is basically is_valid = False when values are nan. I guess you need a new is_valid property on PumpModelResult using the pump_flags you have defined?

Reply:
Agree, that is probably what we need.